### DISCUSSION TITLE
Strange issues Extruder temperature and Stepper_X

### DISCUSSION THREAD

**Discussion by Kr1sDarnok** [bassamanator/Sovol-SV06-firmware#142] [Q&A] [answered]
Probably (and hopefully) a newbie question: I've swapped the broken mainboard of my SOVOL SV06 for a BTT SKR MINI E3 v.3
I followed Bassamanator's instructions to the letter but for some reason I have 2 issue:
1. The displayed temperature of the extruder is way off (235*C at room temp). Switching the thermistor wires from bed to extruder and v.v. doesn't change anything. The extruder temp keeps the culprit.
2. I get the error Unable to read dmc uart 'stepper x' register IFCNT.
I've checked wires and plugs but all seems fine. I even reflashed Marlin and both issues are gone!
Any suggestions are very welcome!
![Klipper issues](https://github.com/bassamanator/Sovol-SV06-firmware/assets/99897055/b884a6ab-28c4-411b-ae51-3c0911eb4576)

**Comment by Kr1sDarnok** [comment]
Additionally, the screenshot above was made after flashing the BTT released klipper firmware, with an extruder temp of 235. If I use Bassamanator's bin-file from the Misc. folder klipper crashes instantly displaying "MCU 'mcu' shutdown: ADC out of range
This generally occurs when a heater temperature exceeds"

**Comment by bassamanator** [comment]
I'm curious about when you downloaded the config. Right now, github is acting quite strange (maybe only for me?), and regardless of the branch I'm on, it's showing me the instructions for the `master` branch thus telling the user to download the files for `master` (SV06). 

Any chance this is happening to you?

**Reply by Kr1sDarnok** [reply]
It’s strange here too, I noticed GitHub displaying the master readme while looking for the SKR pinout with your notes to double check wires…
The firmware.bin was SKR specific judging by the original name… but I’ll recheck

**Comment by bassamanator** [comment]
Have you added the jumpers to enable sensorless homing? See wiring diagram.

**Reply by Kr1sDarnok** [reply]
Yes, took me some time to source jumpers but I’ve put them on the marked pins. 
Strange thing : I reflashed Marlin to check of thermistors and steppers were working, and I had to remove (!) the jumpers to get sensorless homing to work under Marlin…

**Comment by bassamanator** [comment, marked as answer]
**This is correct**

### Download OSS Klipper Configuration

#### Method 1: Clone the Repository

💡 Make sure `git` is installed (`sudo apt update && sudo apt install git`).

1. `cd ~/printer_data/config`
2. Empty entire `~/printer_data/config` folder.
   - In linux, you can delete files via `rm fileName` and directories via `rmdir directoryName`.
   - In linux, you can list files and folders via `ls -lah`.
3. `git clone -b skr-mini-e3-v3 --single-branch https://github.com/bassamanator/Sovol-SV06-firmware.git .` ⚠️ Don't miss the period!

#### Method 2: Download the ZIP

1. [Download](https://github.com/bassamanator/Sovol-SV06-firmware/archive/refs/heads/skr-mini-e3-v3.zip) the `ZIP` file containing the Klipper configuration.
2. See Step 2 in Method 1.
3. The parent folder in the `ZIP` is `Sovol-SV06-firmware-skr-mini-e3-v3`. This is relevant in the next step.
4. Extract **only** the _contents_ of the parent folder into `~/printer_data/config`.

**Reply by Kr1sDarnok** [reply]
Thank you, I’ll check tomorrow, but I’m pretty sure I’ve cloned the SKR specific files…. TBC!

**Reply by Kr1sDarnok** [reply]
Thanks Bassamanator! Maybe I've screwed up before or the GitHub troubles with it switching back to the master branch automatically caused the error, but for some reason redownloading all configs as per your instructions fixed the issue!
You're te best!  Keep up the good work.

**Reply by bassamanator** [reply]
Likely github. All the best!

**Reply by bassamanator** [reply]
Quick update: [other](https://github.com/orgs/community/discussions/101835) Github users were experiencing this issue as well, appears to be fixed now.
